File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXHBox.php
<?php
namespace php\gui\layout;

use php\gui\UXNode;

/**
 * Class UXHBox
 * @package php\gui\layout
 * @packages gui, javafx
 */
class UXHBox extends UXPane
{
    /**
     * TOP_LEFT, TOP_CENTER, TOP_RIGHT, CENTER_LEFT, ... CENTER, ... BOTTOM_RIGHT,
     * BASELINE_LEFT, BASELINE_CENTER, BASELINE_RIGHT
     * @var string
     */
    public $alignment;

    /**
     * @var float
     */
    public $spacing;

    /**
     * @var bool
     */
    public $fillHeight = true;

    /**
     * @param UXNode[] $nodes [optional]
     * @param int $spacing
     */
    public function __construct(array $nodes, $spacing = 0)
    {
    }

    public function requestLayout()
    {
    }

    /**
     * @param UXNode $node
     * @param string $value ALWAYS, SOMETIMES, NEVER
     */
    static function setHgrow(UXNode $node, $value)
    {
    }

    static function getHgrow(UXNode $node)
    {
    }

    /**
     * @param UXNode $node
     * @param double[]|double $margins
     */
    static public function setMargin(UXNode $node, $margins)
    {
    }
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXHBox.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Node;
import javafx.scene.layout.HBox;
import javafx.scene.layout.Priority;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

import java.util.List;

@Reflection.Name(JavaFXExtension.NS + "layout\\UXHBox")
public class UXHBox<T extends HBox> extends UXPane<T> {
    interface WrappedInterface {
        @Property double spacing();
        @Property Pos alignment();
        @Property boolean fillHeight();

        void requestLayout();
    }

    public UXHBox(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXHBox(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new HBox();
    }

    @Signature
    public void __construct(List<Node> children) {
        __construct(children, 0);
    }

    @Signature
    public void __construct(List<Node> children, double spacing) {
        __wrappedObject = new HBox(children.toArray(new Node[children.size()]));
        getWrappedObject().setSpacing(spacing);
    }

    @Signature
    public static void setHgrow(Node node, @Nullable Priority value) {
        HBox.setHgrow(node, value);
    }

    @Signature
    public static Priority getHgrow(Node node) {
        return HBox.getHgrow(node);
    }

    @Signature
    public static void setMargin(Node node, Insets insets) {
        HBox.setMargin(node, insets);
    }
}
